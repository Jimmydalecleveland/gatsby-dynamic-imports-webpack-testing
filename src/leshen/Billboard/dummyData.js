import lifestyle from '../../images/split-content-lifestyle.jpg'
import lifestyleFull from '../../images/billboard-lifestyle-full.jpg'

export const FLUID_IMAGE_DATA = {
  context: {
    custom: {
      alt: 'Alt Text',
    },
  },
  fluid: {
    base64:
      'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAPABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAQFAQL/xAAVAQEBAAAAAAAAAAAAAAAAAAABAv/aAAwDAQACEAMQAAABY4VxKBIGf//EABoQAAIDAQEAAAAAAAAAAAAAAAABAgMREjL/2gAIAQEAAQUCrjpP1qE3lksOz//EABYRAQEBAAAAAAAAAAAAAAAAAAAREv/aAAgBAwEBPwHSv//EABURAQEAAAAAAAAAAAAAAAAAAAAR/9oACAECAQE/AYj/xAAcEAACAQUBAAAAAAAAAAAAAAAAATECEBJBUWH/2gAIAQEABj8C6JUySZ6F7b//xAAcEAACAgIDAAAAAAAAAAAAAAABEQAxIVEQQZH/2gAIAQEAAT8hIHQaigsQF4gXT2NWQVuEdhW3x//aAAwDAQACAAMAAAAQ4+//xAAXEQADAQAAAAAAAAAAAAAAAAAAAREh/9oACAEDAQE/EMVUo//EABYRAQEBAAAAAAAAAAAAAAAAAAARUf/aAAgBAgEBPxDaH//EAB4QAQACAgEFAAAAAAAAAAAAAAERIQAxcUFRYZGx/9oACAEBAAE/EEKI60I86xyRZUEjqPvNJHwy9miG2LXgZRab+9sUtAHGf//Z',
    src: lifestyle,
    srcSet: lifestyle,
    aspectRatio: 1.3524904214559388,
    sizes: '(max-width: 706px) 100vw, 706px',
    presentationWidth: 2560,
  },
}

export const FLUID_IMAGE_DATA_FULL = {
  context: {
    custom: {
      alt: 'Alt Text',
    },
  },
  fluid: {
    base64:
      'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAGABQDASIAAhEBAxEB/8QAFgABAQEAAAAAAAAAAAAAAAAAAAMC/8QAFQEBAQAAAAAAAAAAAAAAAAAAAAH/2gAMAwEAAhADEAAAAdULJA//xAAZEAACAwEAAAAAAAAAAAAAAAABAgADBBL/2gAIAQEAAQUCTYTBa1kfXZ3/AP/EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQMBAT8BP//EABURAQEAAAAAAAAAAAAAAAAAABAR/9oACAECAQE/AYf/xAAcEAACAQUBAAAAAAAAAAAAAAAAEQECEiExQXH/2gAIAQEABj8C0i6ZxTwlJeH/xAAZEAADAQEBAAAAAAAAAAAAAAABESEAQVH/2gAIAQEAAT8hN23nNcLWQCawQJjHn//aAAwDAQACAAMAAAAQBB//xAAWEQEBAQAAAAAAAAAAAAAAAAABEBH/2gAIAQMBAT8QXJ//xAAWEQEBAQAAAAAAAAAAAAAAAAABABH/2gAIAQIBAT8QA2X/xAAaEAEAAgMBAAAAAAAAAAAAAAABACERMWGR/9oACAEBAAE/EGQY0bPrcwQjrko72algtuO3P//Z',
    src: lifestyleFull,
    srcSet: lifestyleFull,
    aspectRatio: 3.116883116883117,
    sizes: '(max-width: 800px) 100vw, 800px',
    presentationWidth: 2560,
  },
}
