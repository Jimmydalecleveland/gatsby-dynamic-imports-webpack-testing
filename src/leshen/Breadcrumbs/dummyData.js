export const CRUMB_DATA = [
  {
    shortName: 'Home',
    path: '/',
  },
  {
    shortName: 'Internet',
    path: '/internet',
  },
  {
    shortName: 'Gigabit',
    path: '/gigabit',
  },
  {
    shortName: 'Gigabit Plans',
    path: '/gigabit-plans',
  },
]
