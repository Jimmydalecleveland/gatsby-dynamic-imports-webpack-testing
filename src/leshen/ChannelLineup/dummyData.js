const CHANNEL_LINEUP_DATA = {
  channels: {
    edges: [
      {
        node: {
          name: 'History Channel',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'history-channel',
        },
      },
      {
        node: {
          name: 'Nickelodeon ',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'nickelodeon',
        },
      },
      {
        node: {
          name: 'FX',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'fx',
        },
      },
      {
        node: {
          name: 'ESPN',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'espn',
        },
      },
      {
        node: {
          name: 'FOX',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'fox',
        },
      },
      {
        node: {
          name: 'CNN',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'cnn',
        },
      },
      {
        node: {
          name: 'Cartoon Network',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'cartoon-network',
        },
      },
      {
        node: {
          name: 'Comedy Central',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'comedy-central',
          url: '/tv/channels/comedy-central',
        },
      },
      {
        node: {
          name: 'Nickelodeon / Nick At Nite - West (NICKW)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'nickelodeon',
        },
      },
      {
        node: {
          name: 'Turner Network Television (TNT)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
          ],
          brandy_id: 'tnt',
          url: '/tv/channels/tnt',
        },
      },
      {
        node: {
          name: 'TV Land (TVLND)',
          packageAvailability: ['dn-americastop120+', 'dn-americastop250'],
          brandy_id: 'tvland',
        },
      },
      {
        node: {
          name: 'E! Entertainment Television (E!)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
          ],
          brandy_id: 'e',
        },
      },
      {
        node: {
          name: 'Syfy Channel (SYFY)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'syfy',
        },
      },
      {
        node: {
          name: 'Eternal Word Television Network (EWTN)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'ewtn',
        },
      },
    ],
  },
  packages: {
    edges: [
      {
        node: {
          name: "America's Top 120",
          brandy_id: 'dn-americastop120',
        },
      },
      {
        node: {
          name: "America's Top 120+",
          brandy_id: 'dn-americastop120+',
        },
      },
      {
        node: {
          name: "America's Top 200",
          brandy_id: 'dn-americastop200',
        },
      },
      {
        node: {
          name: "America's Top 250",
          brandy_id: 'dn-americastop250',
        },
      },
    ],
  },
}

export default CHANNEL_LINEUP_DATA
