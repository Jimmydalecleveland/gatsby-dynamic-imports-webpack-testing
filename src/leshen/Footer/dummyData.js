export const FOOTER_NAV_ITEMS = [
  {
    text: 'Services',
    href: false,
    subnav: [
      {
        text: 'Clearlink Internet',
        href: '#',
      },
      {
        text: 'FiOS Internet',
        href: '#',
      },
      {
        text: 'Bundles',
        href: '#',
      },
      {
        text: 'Frontier TV',
        href: '#',
      },
      {
        text: 'Dish TV',
        href: '#',
      },
      {
        text: 'Clearlink Phone',
        href: '#',
      },
      {
        text: 'Business',
        href: '#',
      },
    ],
  },
  {
    text: 'About',
    href: false,
    subnav: [
      {
        text: 'Store Locations',
        href: '#',
      },
      {
        text: 'Media Center',
        href: '#',
      },
      {
        text: 'Scholarship',
        href: '#',
      },
      {
        text: 'Contact Us',
        href: '#',
      },
    ],
  },
  {
    text: 'Español',
    href: false,
    subnav: [
      {
        text: 'Internet',
        href: '#',
      },
      {
        text: 'FiOS Internet',
        href: '#',
      },
      {
        text: 'DishLATINO',
        href: '#',
      },
      {
        text: 'Teléfono',
        href: '#',
      },
      {
        text: 'Paquetes',
        href: '#',
      },
    ],
  },
  {
    text: 'Terms',
    href: false,
    subnav: [
      {
        text: 'Legal',
        href: '#',
      },
      {
        text: 'Privacy Policy',
        href: '#',
      },
      {
        text: 'Terms of Service',
        href: '#',
      },
    ],
  },
]

export const DISCLAIMERS = [
  '©2019 Internet Service Partners. All Rights Reserved. The name CenturyLink and the pathways logo are trademarks of CenturyLink.',
  '*Geographic and service restrictions apply to AT&T services. Call to see if you qualify. Fiber optics may apply to all or part of the network, depending on your location.',
  '© 2019 AT&T Intellectual Property. AT&T, the AT&T logo and all other AT&T marks contained herein are trademarks of AT&T Intellectual Property and/or AT&T affiliated companies. All rights reserved. Subsidiaries and affiliates of AT&T Inc. provide products and services under the AT&T brand.',
]
