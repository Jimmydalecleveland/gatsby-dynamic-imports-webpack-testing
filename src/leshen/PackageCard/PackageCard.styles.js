import styled from "@emotion/styled"
import { rem } from "polished"
import Brandy from "../Brandy"
import List from "../List"

export const Animator = styled.div`
  margin-left: auto;
  margin-right: auto;

  > * {
    transition: all 0.3s ease-in;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.lg}) {
    &:hover {
      cursor: pointer;

      > * {
        transform: translateY(-8px);
        box-shadow: 0 0 0 1px ${({ theme }) => theme.colors.primary.base.hex},
          0 8px 10px ${({ theme }) => theme.colors.light.darker.hex};
        border-color: ${({ theme }) => theme.colors.primary.dark.hex};

        button {
          background-color: ${({ theme }) => theme.colors.primary.dark.hex};
        }
      }
    }
  }
`

export const Bullets = styled(List)`
  margin: ${({ theme }) => theme.spacing.sm};
`

export const Disclaimer = styled(Brandy)`
  margin: ${({ theme }) => theme.spacing.sm};
`

// TODO: This needs to be refactored to new style approach
// Overwriting the LinkButton this way to display none the icon is a temporary
// adjustment to get away from flex-box nonesense when the button is 100% width
export const PackageCard = styled.div`
  box-shadow: 0 4px 8px ${({ theme }) => theme.colors.light.darker.hex};
  border-top: ${({ theme }) =>
    `solid ${theme.spacing.xxs} ${theme.colors.primary.base.hex}`};
  background-color: ${({ theme }) => theme.colors.light.lighter.hex};
  border-radius: ${({ theme }) => theme.radius};
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100%;
  width: 100%;
  max-width: 320px;
  margin-left: auto;
  margin-right: auto;
  text-align: center;

  a {
    align-self: stretch;
    margin: ${({ theme }) => `${theme.spacing.lg} 0 0`};

    button {
      width: 100%;
      border-top-left-radius: 0;
      border-top-right-radius: 0;
      justify-content: center;

      svg {
        display: none;
      }
    }
  }

  .label {
    transform: ${({ theme }) =>
      `translateY(calc(-100% - ${theme.spacing.xxs}))`};
    padding: ${({ theme }) => `${theme.spacing.min} ${theme.spacing.lg}`};
    background-color: ${({ theme }) => theme.colors.primary.dark.hex};
    font-size: ${({ theme }) => rem(theme.fonts.body.xxs.size)};
    line-height: ${({ theme }) => rem(theme.fonts.body.xxs.lineHeight)};
    color: ${({ theme }) => theme.colors.primary.dark.readable};
    position: absolute;
    margin: 0;
    border-top-left-radius: 4px;
    border-top-right-radius: 4px;
    font-weight: bold;
    letter-spacing: 1px;
    text-transform: uppercase;
  }

  .title {
    margin: ${({ theme }) => `${theme.spacing.xxs} 0 ${theme.spacing.sm}`};
    padding: ${({ theme }) => `0 ${theme.spacing.lg} 0`};
    font-size: ${({ theme }) => rem(theme.fonts.body.xxs.size)};
    line-height: ${({ theme }) => rem(theme.fonts.body.xxs.lineHeight)};
    font-weight: bold;
    letter-spacing: 1px;
    text-transform: uppercase;
  }

  .details {
    margin: ${({ theme }) => `0 0 ${theme.spacing.sm}`};
    padding: ${({ theme }) => `${theme.spacing.sm} 0`};
    background-color: ${({ theme }) => theme.colors.light.base.hex};
    display: flex;
    justify-content: space-around;
    align-items: stretch;
    font-size: 12px;
    line-height: 14px;
    width: 100%;

    > span {
      padding: ${({ theme }) => `${theme.spacing.min}  ${theme.spacing.xs}`};
      display: flex;
      align-items: center;
      justify-content: center;
      position: relative;
      flex: 1;

      &::after {
        background-color: ${({ theme }) => theme.colors.dark.lightest.hex};
        display: block;
        position: absolute;
        right: 0;
        top: 0;
        content: "";
        width: 1px;
        height: 100%;
      }

      &:last-of-type::after {
        display: none;
      }
    }
  }

  .spacer {
    flex: 1;
  }

  .price {
    margin-bottom: ${({ theme }) => theme.spacing.sm};

    &__wrapper {
      display: inline-block;
    }
  }
`
