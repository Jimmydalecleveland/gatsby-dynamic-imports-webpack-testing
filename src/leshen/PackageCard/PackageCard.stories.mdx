import {
  Story,
  Meta,
  Preview,
  Props,
  Anchor,
} from "@storybook/addon-docs/blocks"

import PackageCard, { ticketID } from "."
import Columns from "../Columns"
import Section from "../Section"
import LinkButton from "../LinkButton"
import { PACKAGE_DATA } from "../PackageProvider/dummyData"
import ComponentProgress from "../ComponentProgress"
import { PACKAGE_CARD } from "../utils/dataTrackingValues"

<Meta
  title="Components|PackageCard"
  component={PackageCard}
  decorators={[
    story => (
      <>
        <ComponentProgress ticketID={ticketID} />
        {story()}
      </>
    ),
  ]}
/>

# Package Card

<Preview>
  <Story name="package card">
    <Section>
      <PackageCard
        label="Best Value"
        packageData={PACKAGE_DATA}
        cta={
          <LinkButton
            to="/test"
            variant="feature"
            data-tracking={PACKAGE_CARD.BUTTON}
          >
            Learn More
          </LinkButton>
        }
      />
    </Section>
  </Story>
</Preview>

## Props

<Props of={PackageCard} />

## Package Card with custom children

When child nodes are passed to `PackageCard`, they are rendered above the main CTA at the bottom. This allows for custom additions after the main information that is commonly found in a package card. In this example the "Terms and Conditions" link has been passed.

<Preview>
  <Story name="package card with custom children">
    <Section>
      <PackageCard
        label="Best Value"
        packageData={PACKAGE_DATA}
        cta={
          <LinkButton
            to="/test"
            variant="feature"
            data-tracking={PACKAGE_CARD.BUTTON}
          >
            Learn More
          </LinkButton>
        }
      >
        <a
          href="/terms"
          style={{
            fontSize: "10px",
            fontWeight: "bold",
            textDecoration: "none",
            color: "#444",
          }}
        >
          Terms and Conditions
        </a>
      </PackageCard>
    </Section>
  </Story>
</Preview>

## Package Card Section

<Preview>
  <Story name="package card section">
    <Section>
      <Columns>
        <PackageCard
          label="Best Value"
          packageData={PACKAGE_DATA}
          cta={
            <LinkButton to="/test" variant="feature">
              Learn More
            </LinkButton>
          }
        />
        <PackageCard
          label="Best Value"
          packageData={PACKAGE_DATA}
          cta={
            <LinkButton to="/test" variant="feature">
              Learn More
            </LinkButton>
          }
        />
        <PackageCard
          label="Best Value"
          packageData={PACKAGE_DATA}
          cta={
            <LinkButton to="/test" variant="feature">
              Learn More
            </LinkButton>
          }
        />
      </Columns>
    </Section>
  </Story>
</Preview>
