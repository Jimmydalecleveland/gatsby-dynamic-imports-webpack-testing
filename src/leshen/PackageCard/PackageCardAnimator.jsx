import React from 'react'
import PropTypes from 'prop-types'
import * as Styled from './PackageCard.styles'

const PackageCardAnimator = ({ children }) => (
  <Styled.Animator>{children}</Styled.Animator>
)

PackageCardAnimator.propTypes = {
  children: PropTypes.node.isRequired,
}
