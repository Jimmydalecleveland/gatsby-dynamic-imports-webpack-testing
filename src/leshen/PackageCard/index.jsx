import React from 'react'
import PropTypes from 'prop-types'
import { Done } from 'emotion-icons/material'

import Typography from '../Typography'
import Brandy from '../Brandy'
import ListItem from '../List/ListItem'
import Price from '../Price'
import * as Styled from './PackageCard.styles'

const PackageCard = ({ label, packageData, cta, children }) => (
  <div>
    <Styled.PackageCard>
      {label && (
        <Typography variant="small" className="label">
          {label}
        </Typography>
      )}

      {packageData.name && (
        <Typography variant="h5" className="title">
          {packageData.name}
        </Typography>
      )}
      {packageData.details && (
        <div className="details">
          {packageData.details.map((detail) => (
            <span key={detail}>{detail}</span>
          ))}
        </div>
      )}
      <Price className="price" {...packageData.price}>
        {packageData.price.disclaimer && (
          <Styled.Disclaimer
            symbol={packageData.price.disclaimer.symbol}
            text={packageData.price.disclaimer.text}
            variant="legal"
          />
        )}
      </Price>
      {packageData.bullets && (
        <Styled.Bullets>
          {packageData.bullets.map((bullet) => (
            <ListItem key={bullet.text} icon={<Done />}>
              <Brandy text={bullet.text} variant="body" />
              {bullet.disclaimer && (
                <Brandy
                  symbol={bullet.disclaimer.symbol}
                  text={bullet.disclaimer.text}
                  variant="legal"
                />
              )}
            </ListItem>
          ))}
        </Styled.Bullets>
      )}
      {children}

      <div className="spacer" />
      {cta && cta}
    </Styled.PackageCard>
  </div>
)

PackageCard.defaultProps = {
  label: '',
  cta: null,
  children: null,
}

PackageCard.propTypes = {
  label: PropTypes.string,
  packageData: PropTypes.shape({
    name: PropTypes.string,
    details: PropTypes.arrayOf(PropTypes.string),
    bullets: PropTypes.arrayOf(
      PropTypes.shape({
        text: PropTypes.string.isRequired,
        disclaimer: PropTypes.shape({
          symbol: PropTypes.string,
          text: PropTypes.string.isRequired,
        }),
      })
    ),
    price: PropTypes.shape({
      amount: PropTypes.string.isRequired,
      disclaimer: PropTypes.shape({
        symbol: PropTypes.string,
        text: PropTypes.string.isRequired,
      }),
    }),
    disclaimer: PropTypes.string,
  }).isRequired,
  cta: PropTypes.element,
  children: PropTypes.node,
}

export default PackageCard

export const ticketID = 'CMS-169'
