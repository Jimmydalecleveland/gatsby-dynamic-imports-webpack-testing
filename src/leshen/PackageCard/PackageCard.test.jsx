import React from 'react'
import { render } from 'setup-tests'
import PackageCard from '.'
import { PACKAGE_DATA } from '../PackageProvider/dummyData'

describe('PackageCard', () => {
  test('PackageCard should render', () => {
    const { container } = render(
      <PackageCard label="Best Value" packageData={PACKAGE_DATA} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('should render with children if passed', () => {
    const { getByText } = render(
      <PackageCard label="Best Value" packageData={PACKAGE_DATA}>
        Custom Children
      </PackageCard>
    )
    getByText(/custom children/i)
  })
})
