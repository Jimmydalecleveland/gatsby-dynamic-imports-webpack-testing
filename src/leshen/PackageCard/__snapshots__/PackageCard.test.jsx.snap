// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PackageCard PackageCard should render 1`] = `
.emotion-38 {
  box-shadow: 0 4px 8px #d8dcdf;
  border-top: solid 8px #0f6bc0;
  background-color: #fff;
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  max-width: 320px;
  margin-left: auto;
  margin-right: auto;
  text-align: center;
}

.emotion-38 .ejczsmh0 {
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  margin: 32px 0 0;
}

.emotion-38 .ejczsmh0 button {
  width: 100%;
  border-top-left-radius: 0;
  border-top-right-radius: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-38 .ejczsmh0 button svg {
  display: none;
}

.emotion-38 .label {
  -webkit-transform: translateY(calc(-100% - 8px));
  -ms-transform: translateY(calc(-100% - 8px));
  transform: translateY(calc(-100% - 8px));
  padding: 4px 32px;
  background-color: #125b9f;
  font-size: 0.625rem;
  line-height: 0.75rem;
  color: #fff;
  position: absolute;
  margin: 0;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  font-weight: bold;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  text-transform: uppercase;
}

.emotion-38 .title {
  margin: 8px 0 16px;
  padding: 0 32px 0;
  font-size: 0.625rem;
  line-height: 0.75rem;
  font-weight: bold;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  text-transform: uppercase;
}

.emotion-38 .details {
  margin: 0 0 16px;
  padding: 16px 0;
  background-color: #f4f5f6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  font-size: 12px;
  line-height: 14px;
  width: 100%;
}

.emotion-38 .details > span {
  padding: 4px 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-38 .details > span::after {
  background-color: #5b636d;
  display: block;
  position: absolute;
  right: 0;
  top: 0;
  content: '';
  width: 1px;
  height: 100%;
}

.emotion-38 .details > span:last-of-type::after {
  display: none;
}

.emotion-38 .spacer {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-38 .price {
  margin-bottom: 16px;
}

.emotion-38 .price__wrapper {
  display: inline-block;
}

.emotion-0 {
  font-size: 0.875rem;
  line-height: 1.375rem;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-0 {
    width: 100%;
  }
}

.emotion-2 {
  font-weight: 500;
  font-size: calc(0.5555555555555556vw + 17.333333333333332px);
  line-height: calc(1.1111111111111112vw + 18.666666666666668px);
}

@media (max-width:480px) {
  .emotion-2 {
    font-size: 20px;
  }
}

@media (min-width:1200px) {
  .emotion-2 {
    font-size: 24px;
  }
}

@media (max-width:480px) {
  .emotion-2 {
    line-height: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-2 {
    line-height: 32px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-2 {
    width: 100%;
  }
}

.emotion-12 {
  display: inline-grid;
  grid-template-areas: "prefix prefix prefix prefix" "currency dollars change duration" "suffix suffix suffix suffix";
  font-size: 3.25rem;
}

.emotion-4 {
  grid-area: currency;
  font-size: 0.35em;
  line-height: 1;
}

.emotion-6 {
  grid-area: dollars;
  line-height: 0.7;
}

.emotion-8 {
  grid-area: change;
  font-size: 1em;
  line-height: 0.7;
}

.emotion-8::before {
  display: inline-block;
  content: '.';
}

.emotion-10 {
  grid-area: duration;
  font-size: 0.4em;
  line-height: 1;
}

.emotion-15 {
  font-size: 0.625rem;
  line-height: 0.75rem;
  opacity: 0.8;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-15 {
    width: 100%;
  }
}

.emotion-36 {
  display: inline-block;
  text-align: left;
  list-style: none;
  margin: 16px;
}

.emotion-20 {
  margin-bottom: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-20 > div {
  margin-left: 8px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-20 > svg {
  margin: 0;
}

.emotion-20 h5 {
  margin-bottom: 0;
}

.emotion-20 p {
  margin: 0;
}

.emotion-18 {
  font-size: 1rem;
  line-height: 1.375rem;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-18 {
    width: 100%;
  }
}

<div>
  <div
    class="emotion-38 emotion-39"
  >
    <p
      class="label emotion-0 emotion-1"
    >
      Best Value
    </p>
    <h5
      class="title emotion-2 emotion-3"
    >
      DIRECTVÂ® Ultimate All-Included + INTERNET + HOME PHONE
    </h5>
    <div
      class="details"
    >
      <span>
        300 Channels
      </span>
      <span>
        30Mbps Internet
      </span>
      <span>
        Home Phone
      </span>
    </div>
    <div
      aria-label=" $79.99 per /mo"
      class="price emotion-12 emotion-13"
    >
      <span
        class="emotion-4 emotion-5"
      >
        $
      </span>
      <span
        class="emotion-6 emotion-7"
      >
        79
      </span>
      <span
        class="emotion-8 emotion-9"
      >
        99
      </span>
      <span
        class="emotion-10 emotion-11"
      >
        /mo
      </span>
    </div>
    <p
      class="emotion-14 emotion-15 emotion-1"
    >
      For 12 months with 24-mo. DIRECTV agreement and combined billing. Price includes DIRECTV ULTIMATE All- Included Package ($60.00/mo.)
    </p>
    <ul
      class="emotion-35 emotion-36 emotion-37"
    >
      <li
        class="emotion-20 emotion-21"
      >
        <span>
          Done icon Mock
        </span>
        <div>
          <p
            class="emotion-14 emotion-18 emotion-1"
          >
            Speeds up to 20 Mbps
          </p>
        </div>
      </li>
      <li
        class="emotion-20 emotion-21"
      >
        <span>
          Done icon Mock
        </span>
        <div>
          <p
            class="emotion-14 emotion-18 emotion-1"
          >
            No contract
          </p>
        </div>
      </li>
      <li
        class="emotion-20 emotion-21"
      >
        <span>
          Done icon Mock
        </span>
        <div>
          <p
            class="emotion-14 emotion-18 emotion-1"
          >
            No rate hike
          </p>
          <p
            class="emotion-14 emotion-15 emotion-1"
          >
            For 12 months with 24-mo. DIRECTV agreement and combined billing. Price includes DIRECTV ULTIMATE All- Included Package ($60.00/mo.)
          </p>
        </div>
      </li>
    </ul>
    <div
      class="spacer"
    />
  </div>
</div>
`;
